mapboxcli/scripts/directions.py

```diff
@@ -7,15 +7,17 @@
 
 @click.command(short_help="Routing between waypoints.")
 @click.argument('waypoints', nargs=-1)
-@click.option('--profile', default="mapbox.driving", type=click.Choice([
-              'mapbox.driving', 'mapbox.walking', 'mapbox.cycling']),
-              help="Mapbox direction profile id")
+@click.option('--profile', default="mapbox.driving",
+              help="Mapbox direction profile id, must be one of {0}".format(
+                  mapbox.Directions().valid_profiles))
 @click.option('--alternatives/--no-alternatives', default=True,
               help="Generate alternative routes?")
-@click.option('--instructions', default="text", type=click.Choice(["text", "html"]),
-              help="Format for route instructions")
-@click.option('--geometry', default="geojson", type=click.Choice([
-              'geojson', 'polyline', 'false']), help="Geometry encoding")
+@click.option('--instructions', default="text",
+              help="Format for route instructions, must be one of {0}".format(
+                  mapbox.Directions().valid_instruction_formats))
+@click.option('--geometry', default="geojson",
+              help="Geometry encoding, must be one of {0}".format(
+                  mapbox.Directions().valid_geom_encoding))
```

Comment:
Same as above. These seem natural for choice, unless I'm missing something (it's late and it's possible).